### PR TITLE
bpo-44652: Preserve natural order of args in the union type.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
https://bugs.python.org/issue44652

### CODE REVIEW

File: Lib/test/test_types.py [hunk 1]
```diff
@@ -706,11 +711,35 @@ def test_or_type_operator_with_TypeVar(self):
...
+        check((int | str) | (str | int), (int, str))
+        check(typing.Union[int, str] | list, (int, str, list))
+        check(int | typing.Union[str, list], (int, str, list))
+        check((int | str) | (list | int), (int, str, list))
+        check((int | str) | typing.Union[list, int], (int, str, list))
+        check(typing.Union[int, str] | (list | int), (int, str, list))
+        check((str | int) | (int | list), (str, int, list))
+        check((str | int) | typing.Union[int, list], (str, int, list))
+        check(typing.Union[str, int] | (int | list), (str, int, list))
+        return
```
Comment (Fidget-Spinner) [inline Lib/test/test_types.py]: What's this for?

File: Objects/unionobject.c [hunk 2]
```diff
@@ -260,8 +260,8 @@ dedup_and_flatten_args(PyObject* args)
     for (Py_ssize_t i = 0; i < arg_length; i++) {
         int is_duplicate = 0;
         PyObject* i_element = PyTuple_GET_ITEM(args, i);
-        for (Py_ssize_t j = i + 1; j < arg_length; j++) {
-            PyObject* j_element = PyTuple_GET_ITEM(args, j);
+        for (Py_ssize_t j = 0; j < added_items; j++) {
```
Comment (Fidget-Spinner) [inline Objects/unionobject.c:263]: Just noticed this should be faster too (for a small number of unique members :).

**Review by Fidget-Spinner — COMMENTED** [review]
LGTM. Just one comment below.

File: Lib/test/test_types.py (hunk 1 above)
Comment (serhiy-storchaka) [reply Lib/test/test_types.py]: Good catch! I forgot to remove it.

**Review by Fidget-Spinner — APPROVED** [review]
LG. Windows azure pipelines `test_asyncio` is failing which is unrelated.

**Pull request merged**